android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/cookiebanners/ReportSiteDomainsRepository.kt

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.cookiebanners
+
+import android.content.Context
+import android.util.Log
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.engine.gecko.cookiebanners.ReportSiteDomainsRepository.PreferencesKeys.REPORT_SITE_DOMAINS
+import java.io.IOException
+
+/**
+ * A repository to save reported site domains with the datastore API.
+ */
+class ReportSiteDomainsRepository(
+    private val context: Context,
+) {
+    private val Context.dataStore by preferencesDataStore(
+        name = REPORT_SITE_DOMAINS_REPOSITORY_NAME,
+    )
+
+    companion object {
+        const val TAG: String = "ReportSiteDomainsRep"
+        const val SEPARATOR = "@<;>@"
+        const val REPORT_SITE_DOMAINS_REPOSITORY_NAME = "report_site_domains_preferences"
+        const val PREFERENCE_KEY_NAME = "report_site_domains"
+    }
+
+    private object PreferencesKeys {
+        val REPORT_SITE_DOMAINS = stringPreferencesKey(PREFERENCE_KEY_NAME)
+    }
+
+    suspend fun isSiteDomainReported(siteDomain: String) =
+        context.dataStore.data
```

Comment:
For testing, you could provide this dependency via the constructor this way you can replace it a test time with a `mock` or `spy` :)

Reply:
Done